src/Model/Model.java

```diff
@@ -301,15 +242,133 @@ public void addAllAddableGamesToCommandLists(String userName) {
         }
     }
 
+    public EndTurnResult sendAdvanceTurnCommands(String endTurnPlayer){
+        EndTurnResult result = new EndTurnResult();
+        Game game = getAssociatedGame(endTurnPlayer);
+        String nextPlayer = game.getNextTurn(endTurnPlayer);
+        boolean isLastTurn = game.isLastTurn(endTurnPlayer);
+
+        AdvanceTurnCommand advanceTurn = new AdvanceTurnCommand();
+        advanceTurn.setLastTurn(isLastTurn);
+        advanceTurn.setUsername(nextPlayer);
+
+        CommandData commandData = new CommandData();
+        commandData.setType(ClientCommandType.C_ADVANCE_TURN);
+        commandData.setData(new Gson().toJson(advanceTurn));
+
+        try{
+            addCommandToAllPlayers(game, commandData);
+            result.setSuccess(true);
+            return result;
+        }catch (Exception e){
+            result.setErrorMessage(e.getMessage());
+            result.setSuccess(false);
+            return result;
+        }
+
+    }
+
+    public DrawFaceUpResult drawFaceUp(String player, int index){
+        DrawFaceUpResult result = new DrawFaceUpResult();
+        Game game = getAssociatedGame(player);
+        ArrayList<iCard> cards = game.drawFaceUp(index, player);
+
+        //make Client Commands
+        UpdatePlayerStatsCommand updateStatsCommand = new UpdatePlayerStatsCommand();
+        ArrayList<StatsChange> statsChangeArray = new ArrayList<>();
+        StatsChange statsChange = new StatsChange();
+        statsChange.setAmmount(1);
+        statsChange.setType(StatsChangeType.ADD_TRAIN_CAR_CARDS);
+        statsChangeArray.add(statsChange);
+
+        AccountForTrainCarCardDraw accountForDraws = new AccountForTrainCarCardDraw();
+        accountForDraws.setDeckSize(game.getTrainDeckSize());
+
+        ReplaceOneFaceUpCommand replaceCommand = new ReplaceOneFaceUpCommand();
+        replaceCommand.setCard(cards.get(1));
+        replaceCommand.setIndex(index);
+
+        updateStatsCommand.setUsername(player);
+        updateStatsCommand.setChanges(statsChangeArray);
+
+        CommandData commandData = new CommandData();
```

Comment:
If you're creating 3 commandData objects, you should be consistent with naming, and not have one just be an arbitrary CommandData...